ghcup-based setup fails on GHC 8.4.4

When [trying out `ghcup-jobs >= 8` recently](https://github.com/RyanGlScott/gha-sandbox/commit/cc62eab4b40fe394b5c7f72a40f524d75174c323), I noticed that `ghcup` would fail to install on exactly one version: GHC 8.4.4. Here is an [excerpt from the CI job](https://github.com/RyanGlScott/gha-sandbox/runs/2704517620#step:3:138):

```
[ Info  ] downloading: https://downloads.haskell.org/~ghc/8.4.4/ghc-8.4.4-x86_64-deb9-linux.tar.xz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
 61  146M   61 89.8M    0     0   178M      0 --:--:-- --:--:-- --:--:--  177M
100  146M  100  146M    0     0   205M      0 --:--:-- --:--:-- --:--:--  205M
[ Info  ] verifying digest of: ghc-8.4.4-x86_64-deb9-linux.tar.xz
[ Info  ] Unpacking: ghc-8.4.4-x86_64-deb9-linux.tar.xz to /tmp/ghcup-GHniKk
[ Info  ] Installing GHC (this may take a while)
[ Error ] BuildFailed failed in dir "/tmp/ghcup-GHniKk": NonZeroExit 2 "make" ["install"]
    Check the logs at "/github/home/.ghcup/logs" and the build directory "/tmp/ghcup-GHniKk" for more clues.
    Make sure to clean up "/tmp/ghcup-GHniKk" afterwards.
Error: Process completed with exit code 3.
```

Unfortunately, this doesn't really indicate what the actual issue is, but further digging reveals that this happens due to [this bug](https://gitlab.haskell.org/haskell/ghcup-hs/-/blob/b0522507be6fa991a819aaf22f9a551757380821/README.md#libnuma-required). The bug can be worked around by installing `libnuma-dev`, as demonstrated in [this commit](https://github.com/RyanGlScott/gha-sandbox/commit/bae327937e5eaed77f720af1529f4399de4033ef).

See also https://github.com/haskell/actions/issues/57.

It looks like `libnuma1` alone doesn't suffice, judging from [this build error](https://github.com/RyanGlScott/gha-sandbox/runs/2705040129?check_suite_focus=true#step:19:30):

```
[1 of 1] Compiling Main             ( test/MyLibTest.hs, /__w/gha-sandbox/gha-sandbox/dist-newstyle/build/x86_64-linux/ghc-8.4.4/gha-sandbox-0.1/t/gha-sandbox-test/build/gha-sandbox-test/gha-sandbox-test-tmp/Main.o )
Linking /__w/gha-sandbox/gha-sandbox/dist-newstyle/build/x86_64-linux/ghc-8.4.4/gha-sandbox-0.1/t/gha-sandbox-test/build/gha-sandbox-test/gha-sandbox-test ...
/usr/bin/ld.gold: error: cannot find -lnuma

rts/posix/OSMem.c:327:0: error:
     error: undefined reference to 'mbind'

rts/posix/OSMem.c:606:0: error:
     error: undefined reference to 'numa_num_configured_nodes'

rts/posix/OSMem.c:616:0: error:
     error: undefined reference to 'numa_get_mems_allowed'

rts/posix/OSMem.c:606:0: error:
     error: undefined reference to 'numa_num_configured_nodes'

rts/posix/OSMem.c:621:0: error:
     error: undefined reference to 'numa_bitmask_free'

rts/posix/OSMem.c:606:0: error:
     error: undefined reference to 'numa_num_configured_nodes'
collect2: error: ld returned 1 exit status
`gcc' failed in phase `Linker'. (Exit code: 1)
```

I've opened a PR adding `libnuma-dev` in #517.

--- a/.github/workflows/haskell-ci.yml
+++ b/.github/workflows/haskell-ci.yml
@@ -63,7 +63,7 @@ jobs:
       - name: apt
         run: |
           apt-get update
-          apt-get install -y --no-install-recommends gnupg ca-certificates dirmngr curl git software-properties-common libtinfo5
+          apt-get install -y --no-install-recommends gnupg ca-certificates dirmngr curl git software-properties-common libtinfo5 libnuma-dev
           if [ "${{ matrix.setup-method }}" = ghcup ]; then
             mkdir -p "$HOME/.ghcup/bin"
             curl -sL https://downloads.haskell.org/ghcup/0.1.14.1/x86_64-linux-ghcup-0.1.14.1 > "$HOME/.ghcup/bin/ghcup"

--- a/fixtures/all-versions.github
+++ b/fixtures/all-versions.github
@@ -254,7 +254,7 @@ jobs:
       - name: apt
         run: |
           apt-get update
-          apt-get install -y --no-install-recommends gnupg ca-certificates dirmngr curl git software-properties-common libtinfo5
+          apt-get install -y --no-install-recommends gnupg ca-certificates dirmngr curl git software-properties-common libtinfo5 libnuma-dev
           apt-add-repository -y 'ppa:hvr/ghc'
           if [ $((GHCJSARITH)) -ne 0 ] ; then apt-add-repository -y 'ppa:hvr/ghcjs' ; fi
           if [ $((GHCJSARITH)) -ne 0 ] ; then curl -sSL "https://deb.nodesource.com/gpgkey/nodesource.gpg.key" | apt-key add - ; fi

--- a/fixtures/copy-fields-all.github
+++ b/fixtures/copy-fields-all.github
@@ -171,7 +171,7 @@ jobs:
       - name: apt
         run: |
           apt-get update
-          apt-get install -y --no-install-recommends gnupg ca-certificates dirmngr curl git software-properties-common libtinfo5
+          apt-get install -y --no-install-recommends gnupg ca-certificates dirmngr curl git software-properties-common libtinfo5 libnuma-dev
           apt-add-repository -y 'ppa:hvr/ghc'
           apt-get update
           apt-get install -y "$HCNAME" cabal-install-3.4

--- a/fixtures/copy-fields-none.github
+++ b/fixtures/copy-fields-none.github
@@ -171,7 +171,7 @@ jobs:
       - name: apt
         run: |
           apt-get update
-          apt-get install -y --no-install-recommends gnupg ca-certificates dirmngr curl git software-properties-common libtinfo5
+          apt-get install -y --no-install-recommends gnupg ca-certificates dirmngr curl git software-properties-common libtinfo5 libnuma-dev
           apt-add-repository -y 'ppa:hvr/ghc'
           apt-get update
           apt-get install -y "$HCNAME" cabal-install-3.4

--- a/fixtures/copy-fields-some.github
+++ b/fixtures/copy-fields-some.github
@@ -171,7 +171,7 @@ jobs:
       - name: apt
         run: |
           apt-get update
-          apt-get install -y --no-install-recommends gnupg ca-certificates dirmngr curl git software-properties-common libtinfo5
+          apt-get install -y --no-install-recommends gnupg ca-certificates dirmngr curl git software-properties-common libtinfo5 libnuma-dev
           apt-add-repository -y 'ppa:hvr/ghc'
           apt-get update
           apt-get install -y "$HCNAME" cabal-install-3.4

--- a/fixtures/empty-line.github
+++ b/fixtures/empty-line.github
@@ -171,7 +171,7 @@ jobs:
       - name: apt
         run: |
           apt-get update
-          apt-get install -y --no-install-recommends gnupg ca-certificates dirmngr curl git software-properties-common libtinfo5
+          apt-get install -y --no-install-recommends gnupg ca-certificates dirmngr curl git software-properties-common libtinfo5 libnuma-dev
           apt-add-repository -y 'ppa:hvr/ghc'
           apt-get update
           apt-get install -y "$HCNAME" cabal-install-3.4

--- a/fixtures/enabled-jobs.github
+++ b/fixtures/enabled-jobs.github
@@ -164,7 +164,7 @@ jobs:
       - name: apt
         run: |
           apt-get update
-          apt-get install -y --no-install-recommends gnupg ca-certificates dirmngr curl git software-properties-common libtinfo5
+          apt-get install -y --no-install-recommends gnupg ca-certificates dirmngr curl git software-properties-common libtinfo5 libnuma-dev
           apt-add-repository -y 'ppa:hvr/ghc'
           if [ $((GHCJSARITH)) -ne 0 ] ; then apt-add-repository -y 'ppa:hvr/ghcjs' ; fi
           if [ $((GHCJSARITH)) -ne 0 ] ; then curl -sSL "https://deb.nodesource.com/gpgkey/nodesource.gpg.key" | apt-key add - ; fi

--- a/fixtures/irc-channels.github
+++ b/fixtures/irc-channels.github
@@ -196,7 +196,7 @@ jobs:
       - name: apt
         run: |
           apt-get update
-          apt-get install -y --no-install-recommends gnupg ca-certificates dirmngr curl git software-properties-common libtinfo5
+          apt-get install -y --no-install-recommends gnupg ca-certificates dirmngr curl git software-properties-common libtinfo5 libnuma-dev
           apt-add-repository -y 'ppa:hvr/ghc'
           apt-get update
           apt-get install -y "$HCNAME" cabal-install-3.4

--- a/fixtures/messy.github
+++ b/fixtures/messy.github
@@ -171,7 +171,7 @@ jobs:
       - name: apt
         run: |
           apt-get update
-          apt-get install -y --no-install-recommends gnupg ca-certificates dirmngr curl git software-properties-common libtinfo5
+          apt-get install -y --no-install-recommends gnupg ca-certificates dirmngr curl git software-properties-common libtinfo5 libnuma-dev
           apt-add-repository -y 'ppa:hvr/ghc'
           apt-get update
           apt-get install -y "$HCNAME" cabal-install-3.4 fftw3-dev

--- a/fixtures/psql.github
+++ b/fixtures/psql.github
@@ -177,7 +177,7 @@ jobs:
       - name: apt
         run: |
           apt-get update
-          apt-get install -y --no-install-recommends gnupg ca-certificates dirmngr curl git software-properties-common libtinfo5
+          apt-get install -y --no-install-recommends gnupg ca-certificates dirmngr curl git software-properties-common libtinfo5 libnuma-dev
           apt-add-repository -y 'ppa:hvr/ghc'
           apt-get update
           apt-get install -y "$HCNAME" cabal-install-3.4

--- a/fixtures/travis-patch.github
+++ b/fixtures/travis-patch.github
@@ -171,7 +171,7 @@ jobs:
       - name: apt
         run: |
           apt-get update
-          apt-get install -y --no-install-recommends gnupg ca-certificates dirmngr curl git software-properties-common libtinfo5
+          apt-get install -y --no-install-recommends gnupg ca-certificates dirmngr curl git software-properties-common libtinfo5 libnuma-dev
           apt-add-repository -y 'ppa:hvr/ghc'
           apt-get update
           apt-get install -y "$HCNAME" cabal-install-3.4

--- a/haskell-ci.cabal
+++ b/haskell-ci.cabal
@@ -1,6 +1,6 @@
 cabal-version:      2.2
 name:               haskell-ci
-version:            0.13.20210529
+version:            0.13.20210530
 synopsis:           Cabal package script generator for Travis-CI
 description:
   Script generator (@haskell-ci@) for

--- a/src/HaskellCI/GitHub.hs
+++ b/src/HaskellCI/GitHub.hs
@@ -127,7 +127,9 @@ makeGitHub _argv config@Config {..} gitconfig prj jobs@JobVersions {..} = do
 
         githubRun' "apt" envEnv $ do
             sh "apt-get update"
-            sh "apt-get install -y --no-install-recommends gnupg ca-certificates dirmngr curl git software-properties-common libtinfo5" -- libnuma-dev?
+            -- Installing libnuma-dev is required to work around
+            -- https://gitlab.haskell.org/haskell/ghcup-hs/-/blob/b0522507be6fa991a819aaf22f9a551757380821/README.md#libnuma-required
+            sh "apt-get install -y --no-install-recommends gnupg ca-certificates dirmngr curl git software-properties-common libtinfo5 libnuma-dev"
 
             hvrppa <- runSh $ do
                 sh "apt-add-repository -y 'ppa:hvr/ghc'"

